server/routes/signup/signupHandler.ts

```diff
@@ -15,81 +16,180 @@ export default class Handler {
+      body("maxKidAge")
+        .isNumeric()
+        .notEmpty()
+        .withMessage("Max kid age must be a number"),
+      body("numOfKids")
+        .isNumeric()
+        .notEmpty()
+        .withMessage("Num of kids must be a number"),
+      body("comments")
+        .optional()
+        .isString()
+        .withMessage("comments must be a string"),
+    ];
+  };
+
+  babysitterFieldValidation = () => {
+    return [
+      body("name").isString().notEmpty().withMessage("Name must be a string"),
+      body("email").isEmail().notEmpty().withMessage("Email must be a string"),
+      body("password")
+        .isString()
+        .notEmpty()
+        .withMessage("Password must be a string"),
+      body("city").isString().notEmpty().withMessage("City must be a string"),
+      body("street")
+        .isString()
+        .notEmpty()
+        .withMessage("Street must be a string"),
+      body("experience")
+        .isString()
+        .notEmpty()
+        .withMessage("Experience must be a string"),
+      body("age").isNumeric().notEmpty().withMessage("Age must be a number"),
+      body("phoneNumber")
+        .isString()
+        .notEmpty()
+        .withMessage("Phone number must be a string"),
+      body("gender")
+        .isString()
+        .notEmpty()
+        .withMessage("Gender must be a string"),
+      body("comments")
+        .optional()
+        .isString()
+        .withMessage("comments must be a string"),
+    ];
+  };
+
+  parentSignupValidation = async (
+    email: string,
+    password: string
+  ): Promise<Validation> => {
     const minPasswordLength = 4;
-    const { comments, ...inputs } = req.body;
-
-    const hasNullInput = Object.values(inputs).some(
-      (value) => value === null || value === "" || value === 0
-    );
-
-    if (
-      hasNullInput ||
-      !EmailValidator.validate(req.body.email) ||
-      req.body.password.length < minPasswordLength
-    ) {
-      res
-        .status(401)
-        .json({ message: "Please fill out all the required fields" });
-    } else {
-      next();
+
+    if (password.length < minPasswordLength) {
+      return { isValid: false, message: "Password is too short" };
+    }
+
+    if (!EmailValidator.validate(email)) {
+      return { isValid: false, message: "Email isn't valid" };
     }
-  };
 
-  signupParent = async (req: Request, res: Response) => {
     try {
-      const existingParent = await this.dbHandler.existingParent(
-        req.body.email
-      );
+      const existingParent = await this.dbHandler.existingParent(email);
 
       if (existingParent) {
-        return res.status(401).json({ message: "Email is already in use" });
+        return { isValid: false, message: "Email is already in use" };
       }
     } catch (error) {
       console.log(error);
-      res.status(500).json({ message: "Failed sign up" });
     }
 
-    const { password } = req.body;
+    return { isValid: true };
+  };
+
+  babysitterSignupValidation = async (
```

Comment:
if the validations here are the same as the parent then you can call for both users the same validation function - and then there is not need to create 2 the same